lib/line-numbers.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const callsites = require('callsites');
+const micromatch = require('micromatch');
+const sourceMapSupport = require('source-map-support');
+const flatten = require('lodash/flatten');
+const parseTestSourceInFile = require('./test-ast');
+
+const NUMBER_REGEX = /^-?\d+$/;
+const RANGE_REGEX = /^(?<startGroup>-?\d+)-(?<endGroup>-?\d+)$/;
+const LINE_NUMBERS_REGEX = /^(?:-?\d+(?:-\d+)?,?)+$/;
+const DELIMITER = ':';
+
+const pluralizeIf = (string, condition) => condition ? `${string}s` : string;
+
+class FormatError extends Error {
+	constructor(offender) {
+		const numberPluralized = pluralizeIf('number', offender && !NUMBER_REGEX.test(offender) && offender.length !== 1);
+		super(`Invalid line ${numberPluralized}: \`${offender}\`. Expected comma-separated list of \`[X|Y-Z]\`.`);
+		this.name = 'FormatError';
+	}
+}
+
+const parseNumber = string => parseInt(string, 10);
+const removeAllWhitespace = string => string.replace(/\s/g, '');
+
+const parseLineNumbers = suffix => {
```

Comment:
I still need to check how these errors are presented on the CLI. Especially file paths, which would be absolute values which is jarring.

Reply:
Verdict?

Reply:
☝️ This one remains unanswered

Reply:
I haven't checked yet, sorry 😄 Will resolve for now.